### DISCUSSION TITLE
关于skynet.sleep时间不准确的问题

### DISCUSSION THREAD

**Discussion by uiuizz** [cloudwu/skynet#1876] [General]
skynet.sleep(20)，这样200ms会执行函数。但如果当定时器向服务队列添加消息时，消息队列里面已经有很多了。那么添加的定时消息就会延迟执行，如果想尽量准确200ms执行，有什么好的办法吗

**Comment by CoolDesert** [comment]
btw, 这个你应该放到 discussions 去讨论。

**Comment by huahua132** [comment]
你可以把想尽量准时执行的任务分给其他不怎么忙的服务。

**Reply by uiuizz** [reply]
这个是控制场景scene服务心跳的，这两天我做压测，scene很繁忙。场景心跳如果是必须的，我感觉超时不可避免

**Reply by uiuizz** [reply]
如果队列的消息处理的足够快，超时一点点也可以接受

**Reply by huahua132** [reply]
因为skynet服务是同生共死的，我觉得你的心跳任务处理可以拆分出来用一个服务单独去发

**Reply by uiuizz** [reply]
嗯，还是想办法把scene拆分一下，不过slg大地图不好弄，我看云风有个文章写了怎么拆，他那个直接把地图分成多个服务，看着也挺复杂。还是先把和场景关系不大的拆出去简单些

**Comment by CoolDesert** [comment]
尽可能把服务拆分，提高并行处理效率。

uiuizz ***@***.***>于2024年2月21日 周三21:38写道：

**Reply by uiuizz** [reply]
确实，后面不能设计那种很大的服务
